main/src/library/Reader.flix

```diff
@@ -27,5 +38,19 @@ trait Reader[t] {
     ///
     /// Returns `Err(e)` if some underlying I/O error occurs.
     ///
-    pub def read(b: Array[Int8, r], r: t): Result[IoError, Int32] \ {r, IO}
+    pub def read(b: Array[Int8, r], r: t): Result[IoError, Int32] \ {r, Reader.Eff[t]}
+
+}
+
+instance Reader[InputStream] {
+
+    type Eff = IO
+
+    pub def read(b: Array[Int8, r], r: InputStream): Result[IoError, Int32] \ {r, IO} =
+        try {
+            checked_ecast(Ok(r.read(b)))
```

Comment:
I suppose this is because otherwise we do not get the r effect.

Reply:
I was wondering about this, bc ideally the java `InputStream::read` would be known to flix as having this `r` effect, since it reads from the region. In the flix docs it says:

> The Flix compiler ships with a built-in database that maps classes, constructors, and methods in the Java Class Library to primitive effects.

and maybe it's not possible to add `r` to that mapping bc it's not primitive (and also depends on the flix type of `b` argument)?

Reply:
Its not yet possible. We can only describe primitive effects.